saleor/checkout/fetch.py

```diff
@@ -222,13 +242,31 @@ def fetch_checkout_info(
 ) -> CheckoutInfo:
     """Fetch checkout as CheckoutInfo object."""
 
+    from .utils import get_app_shipping_id
+
     channel = checkout.channel
     shipping_address = checkout.shipping_address
+    shipping_channel_listings = None
+
     shipping_method = checkout.shipping_method
-    shipping_channel_listings = ShippingMethodChannelListing.objects.filter(
-        shipping_method=shipping_method, channel=channel
-    ).first()
-    delivery_method = checkout.collection_point or shipping_method
+    if shipping_method:
+        shipping_channel_listings = ShippingMethodChannelListing.objects.filter(
+            shipping_method=shipping_method, channel=channel
+        ).first()
+        delivery_method: Optional[
+            Union["ShippingMethodData", "Warehouse"]
```

Comment:
I don't think we need typehint here. It's quite huge and makes this block of code hard to read